common/src/KokkosFFT_asserts.hpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: (C) The Kokkos-FFT development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0 WITH LLVM-exception
+
+#ifndef KOKKOSFFT_ASSERTS_HPP
+#define KOKKOSFFT_ASSERTS_HPP
+
+#include <stdexcept>
+#include <sstream>
+#include <string>
+
+#if defined(__cpp_lib_source_location) && __cpp_lib_source_location >= 201907L
+#include <source_location>
+#define KOKKOSFFT_EXPECTS(expression, msg)                            \
```

Comment:
Not a modification from this PR.

I guess the naming is inspired from Kokkos ? In Kokkos, https://github.com/kokkos/kokkos/blob/08ceff92bcf3a828844480bc1e6137eb74028517/core/src/Kokkos_Assert.hpp#L59, these checks are disabled in non-debug, shall we also do that ?

Reply:
Maybe change the name in the future. This check is stricter than Kokkos one in order to avoid random errors from invalid usage of fft libraries.

Reply:
I understand the purpose and agree with it. What do you think of `KOKKOSFFT_THROW_IF` ?

Reply:
That makes sense. 
So, `KOKKOSFFT_THROW_IF` throws if `expression` is satisfied?
Then we need to invert conditionals.

Reply:
> So, `KOKKOSFFT_THROW_IF` throws if `expression` is satisfied? Then we need to invert conditionals.

Yes that would be it.

A second reason to rename is that Kokkos aborts whereas you throw an exception. You let the user a chance to catch the error whereas Kokkos does not.

Reply:
Sure. 
The reason why we do not abort is that many FFT libraries do not abort but throw an exception (or return an error state).

Reply:
Seems ok for me

Reply:
I would like to make the change from `KOKKOSFFT_EXPECTS` to `KOKKOSFFT_THROW_IF` in another PR, since it is big and I may make a mistake. Is it fine with you?

Reply:
Sure!